schedulers/terraform/aws-batch-eks/eks.tf

```diff
@@ -0,0 +1,249 @@
+  }
+  rule {
+    api_groups = ["rbac.authorization.k8s.io"]
+    resources  = ["clusterroles", "clusterrolebindings"]
+    verbs      = ["get", "list"]
+  }
+  depends_on = [module.eks]
+
+}
+
+resource "kubernetes_cluster_role_binding" "batch_cluster_role_binding" {
+  metadata {
+    name = "aws-batch-cluster-role-binding"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = kubernetes_cluster_role.batch_cluster_role.metadata[0].name
+  }
+  subject {
+    kind      = "User"
+    name      = "aws-batch"
+    api_group = "rbac.authorization.k8s.io"
+    namespace = ""
+  }
+  depends_on = [module.eks]
+}
+
+resource "kubernetes_role" "batch_compute_env_role" {
+  metadata {
+    name      = "aws-batch-compute-environment-role"
+    namespace = kubernetes_namespace.doeks_batch_namespace.metadata[0].name
+  }
+
+  rule {
+    api_groups = [""]
+    resources  = ["pods"]
+    verbs      = ["create", "get", "list", "watch", "delete", "patch"]
+  }
+  rule {
+    api_groups = [""]
+    resources  = ["serviceaccounts"]
+    verbs      = ["get", "list"]
+  }
+  rule {
+    api_groups = ["rbac.authorization.k8s.io"]
+    resources  = ["roles", "rolebindings"]
+    verbs      = ["get", "list"]
+  }
+  depends_on = [module.eks]
+
+}
+
+resource "kubernetes_role_binding" "batch_compute_env_role_binding" {
+  metadata {
+    name      = "aws-batch-compute-environment-role-binding"
+    namespace = kubernetes_namespace.doeks_batch_namespace.metadata[0].name
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "Role"
+    name      = kubernetes_role.batch_compute_env_role.metadata[0].name
+  }
+  subject {
+    kind      = "User"
+    name      = "aws-batch"
+    api_group = "rbac.authorization.k8s.io"
+    namespace = ""
+  }
+  depends_on = [module.eks]
+}
+
+# Manage the AWS Batch aws-auth ConfigMap entries
+module "eks_auth" {
+  source  = "terraform-aws-modules/eks/aws//modules/aws-auth"
+  version = "~> 20.0"
+
+  manage_aws_auth_configmap = true
+  aws_auth_roles = [
+    {
+      rolearn  = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/AWSServiceRoleForBatch"
+      username = "aws-batch"
+      groups   = []
+    },
+    {
+      rolearn  = module.eks.eks_managed_node_groups["default"].iam_role_arn
+      username = "system:node:{{EC2PrivateDNSName}}"
+      groups   = ["system:bootstrappers", "system:nodes"]
+    },
+    {
+      rolearn  = aws_iam_role.batch_eks_instance_role.arn
+      username = "system:node:{{EC2PrivateDNSName}}"
+      groups   = ["system:bootstrappers", "system:nodes"]
+    }
+  ]
+  depends_on = [
+    module.eks,
+    kubernetes_namespace.doeks_batch_namespace,
+    kubernetes_cluster_role.batch_cluster_role,
+    kubernetes_cluster_role_binding.batch_cluster_role_binding,
+    kubernetes_role.batch_compute_env_role,
+    kubernetes_role_binding.batch_compute_env_role_binding
+  ]
+}
```

Comment:
You dont need this. you can pass the additional auth roles through variables. Check the EKS TF module for input variables.

Reply:
Can you point me at an example? I am not seeing the input for config maps from the [EKS TF module](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest?tab=inputs). I only see how to do this via the [aws-auth submodule](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest/submodules/aws-auth) which I am using in the above code.